### PR TITLE
[css-env-1] Define safe area inset variables #2629

### PR DESCRIPTION
**Pull request by rebeccahughes**
Define CSS safe area inset variables as part of #2629

### CODE REVIEW

**Comment by rebeccahughes** [comment]
cc: @tabatkins @grorg

File: css-env-1/Overview.bs [hunk 1]
```diff
@@ -77,6 +77,53 @@ tho user agents may define additional undocumented [=environment variables=]:
...
+	Applies to: all elements
+	Value: <<length>>
+	Initial: 0
+	Inherited: yes
+	Media: visual
+	Animatable: no
+</pre>
+
+The safe area insets are four [=environment variables=] that define the top, right, bottom and
+left insets from the edge of the document. Page content that is in this safe area may be
```
Comment (astearns) [inline css-env-1/Overview.bs]: 'document' seems wrong here - shouldn't it be something like 'viewport'?

Comment (astearns) [reply css-env-1/Overview.bs]: And I've been assuming the safe area is the bit inside the insets, so it I think it would be 'content that is outside this safe area' or 'content that is inside these insets'

Comment (rebeccahughes) [reply css-env-1/Overview.bs]: Sounds good - fixed.

File: css-env-1/Overview.bs [hunk 2]
```diff
@@ -77,6 +77,53 @@ tho user agents may define additional undocumented [=environment variables=]:
 
 * Issue: define list of predefined vars
 
+Safe area inset variables {#safe-area-insets}
+------------------------------------------------------------------
+
+<pre class=propdef>
```
Comment (dbaron) [inline css-env-1/Overview.bs]: I think defining these using `propdef` entries doesn't seem right.  The only aspects of the propdef tables that actually seem to apply here are "Name", "Value", and maybe "Media".  Furthermore, using propdef entries will put them in property indexes, which will be very confusing.

I think in the absence of a better mechanism this should probably be something more like a hand-written table, and if a better mechanism is needed it can be developed later.

File: css-env-1/Overview.bs [hunk 3]
```diff
@@ -77,6 +77,53 @@ tho user agents may define additional undocumented [=environment variables=]:
...
+	Value: <<length>>
+	Initial: 0
+	Inherited: yes
+	Media: visual
+	Animatable: no
+</pre>
+
+The safe area insets are four [=environment variables=] that define the top, right, bottom and
+left insets from the edge of the viewport. Page content that is inside these insets may be
+obscured by the shape of the display.
```
Comment (dbaron) [inline css-env-1/Overview.bs]: I think this description could be clearer about what these insets are for to readers not already familiar with the concept, and should probably also have stronger implementation conformance criteria.  I'd suggest rewording to something like the following:

> The safe area insets are four [=environment variables=] that define a rectangle by its top, right, bottom, and left insets from the edge of the viewport.  For rectangular displays, these must all be zero, but for nonrectangular displays they must form a rectangle, chosen by the user agent, such that all content inside the rectangle is visible, and such that reducing any of the insets would cause some content inside of the rectangle to be invisible due to the nonrectangular nature of the display.  This allows authors to limit the layout of essential content to the space inside of the safe area rectangle.

Comment (rebeccahughes) [reply css-env-1/Overview.bs]: Fixed

File: css-env-1/Overview.bs (hunk 2 above)
Comment (rebeccahughes) [reply css-env-1/Overview.bs]: Fixed

**Comment by dbaron** [comment]
Thanks.  This looks good to me.

Since you kept the "Media" entry in the table, I think the one thing that's still remaining is defining what happens when the medium in question doesn't match.  That's beyond the definition of the safe area inset variables.  I suspect a decent answer would be that the variable isn't present.  But it should probably be defined; I filed #2819 to follow up on that.

**Comment by rebeccahughes** [comment]
Thanks!

**Comment by astearns** [comment]
@rebeccahughes do you work at Google?

**Comment by rebeccahughes** [comment]
@astearns Yes I do
